src/v/net/server.h

```diff
@@ -71,6 +71,12 @@ struct config_connection_rate_bindings {
     config::binding<std::vector<ss::sstring>> config_overrides_rate;
 };
 
+struct tcp_keepalive_bindings {
```

Comment:
I wonder if we need a way to keep the KA at the default OS values, as it is today? I guess it would mean that each member would be an optional property. @dotnwat any thoughts?

Reply:
I was considering making all config flags optional and am happy to do so. 

I felt that if it's so important that all apps on your system use the system defaults you might as well set them explicitly in the app config?

Reply:
> I felt that if it's so important that all apps on your system use the system defaults you might as well set them explicitly in the app config?

This is a good point. So let's stick with what we have.

I guess this is more important incase where the system default isn't obtainable with an explicit value: like socket buffer sizes. There if you don't set the size some adaptive algo is used but if you set it explicitly you never get the adaptive algo. That's not the case here.